`quarkus run` failing

### Describe the bug

When running `quarkus run` (or `./mvnw quarkus:run`), the application fails to start, yet when running via `java -jar` the application runs fine.

### Expected behavior

I would expect `quarkus run` (or `./mvnw quarkus:run`) to run my application.

When I `java -jar` my application, I see

```bash
╰─ java -jar target/quarkus-app/quarkus-run.jar                                           
__  ____  __  _____   ___  __ ____  ______ 
 --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ 
 -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   
--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   
2024-08-06 12:55:01,071 WARN  [io.qua.config] (main) Unrecognized configuration key "quarkus.langchain4j.ollama.parasol-chat.model-id" was provided; it will be ignored; verify that the dependency extension for this configuration is set or that you did not make a typo
2024-08-06 12:55:02,061 INFO  [io.quarkus] (main) insurance-app 1.0.0-SNAPSHOT on JVM (powered by Quarkus 3.13.0) started in 1.184s. Listening on: http://0.0.0.0:8005
2024-08-06 12:55:02,061 INFO  [io.quarkus] (main) Profile prod activated. 
2024-08-06 12:55:02,061 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, hibernate-orm-panache, jdbc-h2, langchain4j, langchain4j-ollama, langchain4j-openai, langchain4j-websockets-next, narayana-jta, quinoa, qute, rest, rest-client, rest-client-jackson, rest-jackson, smallrye-context-propagation, vertx, websockets-next]
```

### Actual behavior

The application fails to run:

```bash
╰─ ./mvnw quarkus:run                   
[INFO] Scanning for projects...
[INFO] 
[INFO] ---------------------< org.parasol:insurance-app >----------------------
[INFO] Building insurance-app 1.0.0-SNAPSHOT
[INFO]   from pom.xml
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- quarkus:3.13.0:run (default-cli) @ insurance-app ---
[INFO] ------------------------------------------------------------------------
[INFO] BUILD FAILURE
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  2.259 s
[INFO] Finished at: 2024-08-06T12:46:07-04:00
[INFO] ------------------------------------------------------------------------
[ERROR] Failed to execute goal io.quarkus.platform:quarkus-maven-plugin:3.13.0:run (default-cli) on project insurance-app: Execution default-cli of goal io.quarkus.platform:quarkus-maven-plugin:3.13.0:run failed: io.quarkus.builder.BuildException: Build failure: Build failed due to errors
[ERROR] 	[error]: Build step io.quarkus.arc.deployment.BeanArchiveProcessor#build threw an exception: java.lang.IllegalStateException: Failed to index: io.quarkus.test.TestTransaction
[ERROR] 	at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:168)
[ERROR] 	at io.quarkus.arc.deployment.BeanArchiveProcessor.build(BeanArchiveProcessor.java:94)
[ERROR] 	at java.base/jdk.internal.reflect.DirectMethodHandleAccessor.invoke(DirectMethodHandleAccessor.java:103)
[ERROR] 	at java.base/java.lang.reflect.Method.invoke(Method.java:580)
[ERROR] 	at io.quarkus.deployment.ExtensionLoader$3.execute(ExtensionLoader.java:849)
[ERROR] 	at io.quarkus.builder.BuildContext.run(BuildContext.java:256)
[ERROR] 	at org.jboss.threads.ContextHandler$1.runWith(ContextHandler.java:18)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.doRunWith(EnhancedQueueExecutor.java:2516)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2495)
[ERROR] 	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1521)
[ERROR] 	at java.base/java.lang.Thread.run(Thread.java:1583)
[ERROR] 	at org.jboss.threads.JBossThread.run(JBossThread.java:483)
[ERROR] Caused by: java.lang.IllegalArgumentException: stream cannot be null
[ERROR] 	at org.jboss.jandex.Indexer.indexWithSummary(Indexer.java:2576)
[ERROR] 	at io.quarkus.deployment.index.IndexingUtil.indexClass(IndexingUtil.java:163)
[ERROR] 	... 11 more
[ERROR] 
[ERROR] -> [Help 1]
[ERROR] 
[ERROR] To see the full stack trace of the errors, re-run Maven with the -e switch.
[ERROR] Re-run Maven using the -X switch to enable full debug logging.
[ERROR] 
[ERROR] For more information about the errors and possible solutions, please read the following articles:
[ERROR] [Help 1] http://cwiki.apache.org/confluence/display/MAVEN/PluginExecutionException
```

### How to Reproduce?

Reproducer: https://github.com/rh-rad-ai-roadshow/parasol-insurance/tree/main/app

1. Clone https://github.com/rh-rad-ai-roadshow/parasol-insurance (`main` branch)
2. `cd app`
3. `./mvnw clean package -DskipTests`
4. `./mvnw quarkus:run`
5. Observe error

### Output of `uname -a` or `ver`

`Darwin edeandrea-m1pro 23.6.0 Darwin Kernel Version 23.6.0: Fri Jul  5 17:56:41 PDT 2024; root:xnu-10063.141.1~2/RELEASE_ARM64_T6000 arm64`

### Output of `java -version`

```
openjdk version "21.0.3" 2024-04-16 LTS
OpenJDK Runtime Environment Temurin-21.0.3+9 (build 21.0.3+9-LTS)
OpenJDK 64-Bit Server VM Temurin-21.0.3+9 (build 21.0.3+9-LTS, mixed mode)
```

### Quarkus version or git rev

3.13.0

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

```
Apache Maven 3.9.6 (bc0240f3c744dd6b6ec2920b3cd08dcc295161ae)
Maven home: /Users/edeandre/.m2/wrapper/dists/apache-maven-3.9.6/a53741d1
Java version: 21.0.3, vendor: Eclipse Adoptium, runtime: /Users/edeandre/.sdkman/candidates/java/21.0.3-tem
Default locale: en_US, platform encoding: UTF-8
OS name: "mac os x", version: "14.6", arch: "aarch64", family: "mac"
```

--- a/extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java
+++ b/extensions/narayana-jta/deployment/src/main/java/io/quarkus/narayana/jta/deployment/NarayanaJtaProcessor.java
@@ -204,6 +204,11 @@ class NarayanaJtaProcessor {
     @BuildStep(onlyIf = IsTest.class)
     void testTx(BuildProducer<GeneratedBeanBuildItem> generatedBeanBuildItemBuildProducer,
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+
+        if (!testTransactionOnClassPath()) {
+            return;
+        }
+
         //generate the annotated interceptor with gizmo
         //all the logic is in the parent, but we don't have access to the
         //binding annotation here
@@ -217,6 +222,15 @@ class NarayanaJtaProcessor {
         }
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(TestTransactionInterceptor.class)
                 .addBeanClass(TEST_TRANSACTION).build());
+    }
+
+    private static boolean testTransactionOnClassPath() {
+        try {
+            Class.forName(TEST_TRANSACTION, false, Thread.currentThread().getContextClassLoader());
+            return true;
+        } catch (ClassNotFoundException ignored) {
+            return false;
+        }
     }
 
     @BuildStep

--- a/extensions/panache/hibernate-reactive-panache-common/deployment/src/main/java/io/quarkus/hibernate/reactive/panache/common/deployment/PanacheJpaCommonResourceProcessor.java
+++ b/extensions/panache/hibernate-reactive-panache-common/deployment/src/main/java/io/quarkus/hibernate/reactive/panache/common/deployment/PanacheJpaCommonResourceProcessor.java
@@ -61,6 +61,11 @@ public final class PanacheJpaCommonResourceProcessor {
     @BuildStep(onlyIf = IsTest.class)
     void testTx(BuildProducer<GeneratedBeanBuildItem> generatedBeanBuildItemBuildProducer,
             BuildProducer<AdditionalBeanBuildItem> additionalBeans) {
+
+        if (!testReactiveTransactionOnClassPath()) {
+            return;
+        }
+
         //generate the annotated interceptor with gizmo
         //all the logic is in the parent, but we don't have access to the
         //binding annotation here
@@ -74,6 +79,15 @@ public final class PanacheJpaCommonResourceProcessor {
         }
         additionalBeans.produce(AdditionalBeanBuildItem.builder().addBeanClass(TestReactiveTransactionalInterceptor.class)
                 .addBeanClass(TEST_REACTIVE_TRANSACTION).build());
+    }
+
+    private static boolean testReactiveTransactionOnClassPath() {
+        try {
+            Class.forName(TEST_REACTIVE_TRANSACTION, false, Thread.currentThread().getContextClassLoader());
+            return true;
+        } catch (ClassNotFoundException ignored) {
+            return false;
+        }
     }
 
     @BuildStep

